magento/magento2
# Issue 36049: Catalog Price Rule single product reindex works incorrectly

### Description (*)

Catalog Price Rule single product reindex works incorrectly

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#36049

### Manual testing scenarios (*)

1. Create 2 Websites (W1 and W2)
2. the Product's Attribute (e.g. "Age Group") with the scope "website" and "Use for Promo Rule Conditions" set to "yes"
3. the Product assigned to both websites and having Attribute with option 1 for W1, and option 2 for W2 (e.g. "Children" for W1
and "Adults" for W2).
4. Create a Catalog Price Rule.
5. In the rule as a condition, select option 1 of the Attribute ("Children").
6. Assign the rule to both websites (W1 and W2).
7. Open the Product edit page.
8. Select scope W1 on the Product edit page.
9. Save the product.
10. Open W1 catalog as a User - see the price of the Product.
11. Open the Product edit page again.
12. Select scope W2 on the Product edit page.
13. Open W2 catalog as a User - see the price of the Product.
14. Run command bin/magento indexer:reindex and see the price of the product in both websites  (W1 and W2)

### Questions or comments


### Contribution checklist (*)
